src/Components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,85 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { useContext, useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+import { Todo } from '../../Types/Todo';
+import { TodosContext } from '../../Store';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const { removeTodo, toggleTodo } = useContext(TodosContext);
+
+  const [title, setTitle] = useState(todo.title);
+  const [isEdit, setIsEdit] = useState(false);
+
+  const inpuRef = useRef<HTMLInputElement>(null);
+
+  const handleChangeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+  };
+
+  const handleToggleChange = () => {
+    toggleTodo(todo.id);
+  };
+
+  const handleDoubleClick = () => {
+    setIsEdit(true);
+  };
+
+  const handleEditTodo = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter') {
+      if (title.trim().length !== 0) {
```

Comment:
It will be enough for strings 
```suggestion
      if (title.trim()) {
```